### PR TITLE
QBFT to log warning when using validator contract if genesis extra data has validator

### PR DESCRIPTION
**Pull request by siladu**
Also prevent existing warning from logging when using validator contract with no signers in genesis extra data.

Fixes https://github.com/hyperledger/besu/issues/2744

### CODE REVIEW

**Review by antonydenyer — APPROVED** [review]

**Review by usmansaleem — APPROVED** [review]
LGTM

File: besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java [hunk 1]
```diff
@@ -265,11 +265,44 @@ protected ProtocolSchedule createProtocolSchedule() {
...
+      final BlockHeader genesisBlockHeader) {
+    return genesisConfig
+            .getConfigOptions()
+            .getQbftConfigOptions()
+            .getValidatorContractAddress()
+            .isEmpty()
+        && noSignersExistIn(genesisBlockHeader);
+  }
+
+  private boolean signersExistIn(final BlockHeader genesisBlockHeader) {
```
Comment (jframe) [inline besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java:298]: nit: not convinced this method is adding a lot of value. For the one place this is used I think noSignersExistIn(genesisBlockHeader) is easier to read.

File: besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java [hunk 2]
```diff
@@ -265,11 +265,44 @@ protected ProtocolSchedule createProtocolSchedule() {
...
     }
   }
 
+  private boolean usingValidatorContractModeButSignersExistIn(
+      final BlockHeader genesisBlockHeader) {
+    return genesisConfig
+            .getConfigOptions()
+            .getQbftConfigOptions()
+            .getValidatorContractAddress()
+            .isPresent()
```
Comment (jframe) [inline besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java:284]: Would be nice to remove the duplication between this and usingValidatorBlockHeaderModeButNoSignersIn by creating isValidatorContractMode method

File: besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java (hunk 1 above)
Comment (siladu) [reply besu/src/main/java/org/hyperledger/besu/controller/QbftBesuControllerBuilder.java]: In order to keep the positively worded method, I have changed `signersExistIn` and removed noSignersExistIn (using negation instead)

**Review by jflo — APPROVED** [review]

**Review by jframe — APPROVED** [review]
LGTM

**Pull request merged**